### ISSUE TITLE
unable to stream/no microphone acces

### ISSUE THREAD

**Issue by LithRakoon** [maltejur/discord-screenaudio#52]
When i try to use screenshare it is unable to start a stream:

https://user-images.githubusercontent.com/41258148/192273699-ea709d38-09e7-4afe-9a1c-1ed25c0465e4.mp4

and when i go in to settings to look at my micrphone settings it is unable to acces my microphone with no option to enable it:
![image](https://user-images.githubusercontent.com/41258148/192273648-cf7785ba-0eca-478c-b1b0-7fdbdc05cd2b.png)

**Comment by maltejur** [comment]
Are you using the flatpak?

**Comment by maltejur** [comment]
Have you changed any permissions using Flatseal? Or can you give the program the nescessary permissions using Flatseal?

**Comment by LithRakoon** [comment]
what i showed was a fresh first install of discord-screen audio.
i have tried looking inside flatseal but i could not find anything that would seem to help me, though i could have looked over it

**Comment by maltejur** [comment]
Hm okay. Do other Flatpaks have microphone access on your PC?

**Comment by LithRakoon** [comment]
yes! i have the normal discord client installed as a flatpak and my microphone works on there out of the box

**Comment by Burg-Maladay** [comment]
Not familiar with flakpak, never used it, but I had this exact problem the other day. I solved it by installing pipewire-pulse since I use pulseaudio and restarted my computer. It started working after that. You might have a similar issue where you are missing some audio package?

**Comment by LithRakoon** [comment]
![image](https://user-images.githubusercontent.com/41258148/193066468-c3788615-28a0-45df-a80f-573142bdbef6.png)
i already have that installed

**Comment by maltejur** [comment]
@LithRakoon but you are using PipeWire on your PC, right?

**Comment by Soremwar** [comment]
Same situation here, no matter if installed with flatpak or built in AUR

**Comment by kjarosh** [comment]
Hello, I had the same issue on Arch. When starting screen sharing, nothing happened and the following log was being printed over and over:
```
[userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms
```

I've spent some time on debugging this and I noticed I somehow had both `pipewire` and `pulseaudio` installed... Installing `pipewire-pulse` and uninstalling `pulseaudio` fixed the issue. Verified this on two separate Arch installations.

**Comment by Soremwar** [comment]
Different issue here

Only pipewire installed, full log goes like this
```log
[virtmic] Starting Virtmic with target "[All Desktop Audio]"
[virtmic] Added: discord-screenaudio
[virtmic] Added: discord-screenaudio
[virtmic] Added: discord-screenaudio
[discord] "Uncaught (in promise) NotReadableError: Could not start audio source"
```

**Comment by japareaggae** [comment]
I'm also being affected by this issue. However, I tried discord-screenaudio on a different machine and it worked fine. I tried reinstalling Arch from scratch on both machines, setting up a basic sway environment, and installing discord-screenaudio from the AUR, so both machines have the same configuration, but discord-screenaudio still only works in one of them. There should be no issues with the PipeWire configuration since OBS Studio is able to capture the screen on both machines. Let me know if there's anything I can do to help debug this.

**Comment by maltejur** [comment]
Hm this sounds like it could be a graphics driver problem. Does one of the two machines have Nvdia and do both machines have the latest graphics drivers?

**Comment by japareaggae** [comment]
Both machines are using Intel graphics with the latest drivers. The working one is using an Intel HD Graphics 4600 (integrated with i5-4460 processor) and the non-working one is using an Intel HD Graphics 530 (integrated with i5-6500 processor).

I've ran `inxi -v5` on both machines to view any possible differences:

* Working:  [asrock.txt](https://github.com/maltejur/discord-screenaudio/files/10202909/asrock.txt)
* Non-working: [hp.txt](https://github.com/maltejur/discord-screenaudio/files/10202910/hp.txt)

**Comment by LithRakoon** [comment]
i have figured out what the issue was, for me atleast
By default my pc doesnt have a microphone plugged in,
but when no microphone is plugged in Discord-screenaudio will be unable to access the microphone permissions and itll just not work like in the video shown above.

So what i did as a temporary fix is:
- Put in earbuds/headphone which has microphone
- open Discord-screenaudio
- Start a call
- profit

Now the Discord-screenaudio client should have microphone permissions
This however is not a user bug, this is a bug of Discord-screenaudio. Because when you install the normal discord flatpak it automatically has access to microphone permissions without needing an actual microphone

**Comment by japareaggae** [comment]
Thanks for the suggestion! I tried plugging in a microphone in the machine where discord-screenaudio didn't work, and sure enough, I was able to start streaming!

I think the reason why this is happening is because on the non-working machine, PipeWire won't let you set the analog input as the default input unless something is plugged in to it. On the working machine, PipeWire doesn't seem to care about whether something is plugged in or not.
